bob-backend/src/pearl/core.rs

```diff
@@ -226,6 +226,29 @@ impl BackendStorage for Pearl {
         }
     }
 
+    async fn close_oldest_active_blob(&self) -> Option<usize> {
+        let mut oldest: Option<Holder> = None;
+        for dc in self.disk_controllers.iter() {
+            if let Some(holder) = dc.find_oldest_inactive_holder().await {
+                if holder.end_timestamp()
+                    < oldest
+                        .as_ref()
+                        .map(|h| h.end_timestamp())
+                        .unwrap_or(u64::MAX)
+                {
+                    oldest = Some(holder);
+                }
+            }
+        }
+
+        if let Some(h) = oldest {
+            h.close_active_blob().await;
+            Some(h.index_memory().await)
```

Comment:
Will this give the correct result when the blob is already closed? Maybe `index_memory` call should be right before `close_active_blob`